project/src/test/java/com/google/sps/ServletExampleWithInjectionTest.java

```diff
@@ -0,0 +1,40 @@
+package com.google.sps;
+
+import static org.junit.Assert.*;
+
+import com.google.sps.data.User;
+import com.google.sps.servlets.ServletExampleWithInjection;
+import com.google.sps.utilities.DatabaseWrapper;
+import com.google.sps.utilities.TestInstanceDatabaseWrapper;
+import java.io.*;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.*;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(JUnit4.class)
+public class ServletExampleWithInjectionTest extends Mockito {
+  private final TestInstanceDatabaseWrapper wrapper = new TestInstanceDatabaseWrapper();
```

Comment:
Maybe declare this as its interface instead so we aren't depending on any `TestInstanceDatabaseWrapper`-specific methods - this way we are actually guaranteeing that we're using and testing "real" methods that the real `DatabaseWrapper` has and not something that only exists in test world

Reply:
I think if its an interface we possibly would not be able to do the injection properly because interface implies there is no implementation of any methods?

Reply:
Not that we would instantiate it as in interface (since interfaces can't do that), but `private final DatabaseWrapper wrapper = new TestInstanceDatabaseWrapper()'` where it's of the interface type instead would (I think) guarantee that we can't use methods that are only existent in the TestInstanceDatabaseWrapper class

I don't think it would affect injection, but maybe wait and see what liangqi and hailun say

Reply:
Done - Oh I see, yes that makes sense now. Thank you.